CMake fails to build

Hi all,

I was trying to build the project by command line following the descriptions from README, however the build is failing without providing a build dir.

Steps to Reproduce:
1. Cloned the project
2. `git submodule update --init --recursive`
3. `cmake .`
4. `cmake --build .`

System Info:
- Ubuntu 18.04 LTS
- cmake version 3.10.2

Logs:
```
Scanning dependencies of target update_samples
[ 26%] Generating ValidationSchema files
ValidationSchema.cpp
[ 28%] Generating mock TodaySchema files
TodaySchema.cpp
[ 29%] Generating mock TodaySchema (--separate-files)
[ 31%] Updating sample files
[ 31%] Built target update_samples
Scanning dependencies of target separateschema
make[2]: *** No rule to make target 'samples/separate/TodaySchema.cpp', needed by 'samples/CMakeFiles/separateschema.dir/separate/TodaySchema.cpp.o'.  Stop.
CMakeFiles/Makefile2:1315: recipe for target 'samples/CMakeFiles/separateschema.dir/all' failed
make[1]: *** [samples/CMakeFiles/separateschema.dir/all] Error 2
Makefile:140: recipe for target 'all' failed
make: *** [all] Error 2
```

Also, another problem is regarding the installation outputs following README instructions:
```bash
$ cmake --build . --config Release --target install
...
Install the project...                                                                                                                                                                                                         
-- Install configuration: ""                                                                                                                                                                                
-- Up-to-date: /usr/local/lib/cmake/cppgraphqlgen/cppgraphqlgen-config.cmake                                                                                                                                
-- Up-to-date: /usr/local/lib/cmake/cppgraphqlgen/cppgraphqlgen-config-version.cmake                                                                                                                        
-- Installing: /usr/local/lib/libgraphqljson.a                                                                                                                                                              
-- Installing: /usr/local/lib/libgraphqlpeg.a                                                                                                                                                               
-- Installing: /usr/local/lib/libgraphqlresponse.a                                                                                                                                                          
-- Installing: /usr/local/lib/libgraphqlservice.a                                                                                                                                                           
-- Up-to-date: /usr/local/lib/cmake/cppgraphqlgen/cppgraphqlgen-targets.cmake                                                                                                                               
-- Up-to-date: /usr/local/lib/cmake/cppgraphqlgen/cppgraphqlgen-targets-noconfig.cmake
```

The README command won't setup correctly the configuration inside CMake. On (https://cmake.org/cmake/help/latest/guide/tutorial/index.html#packaging-debug-and-release-step-12)[cmake official doc], their example is a bit different and I think we could improve the README.md with it:
```bash
cmake -DCMAKE_BUILD_TYPE=Release . && cmake --build . --target install
```

If you guys think it is a valid issue, I can already open a PR for that.

[]'s

--- a/README.md
+++ b/README.md
@@ -172,13 +172,14 @@ change the config from `Debug` (default) to `Release`, use another build tool li
 to install the dependencies, remember to specify the `-DCMAKE_TOOLCHAIN_FILE=...` option when you run the initial build
 configuration.
 
-- Configure the build system: `"cmake ."`
+- Create a build directory: `"mkdir build && cd build"`
+- Configure the build system: `"cmake .."`
 - Tell CMake to invoke the build system: `"cmake --build ."` _You can repeat this step to rebuild your changes._
 - CTest comes with CMake and runs the tests: `"ctest ."` _Run this frequently, and make sure it passes before commits._
 
-You can then optionally install the public outputs by running.
-
-- `cmake --build . --config Release --target install` _You probably need to use `sudo` on Unix to do this._
+You can then optionally install the public outputs by configuring it with `Release`:
+- `cmake -DCMAKE_BUILD_TYPE=Release ..`
+- `cmake --build . --target install` _You probably need to use `sudo` on Unix to do this._
 
 ## Interactive tests
 

